common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -451,14 +492,38 @@ public Bundle bundleFromAccounts(@NonNull final List<ICacheRecord> cacheRecords)
         final Bundle resultBundle = new Bundle();
 
         if (cacheRecords != null) {
-            resultBundle.putString(BROKER_ACCOUNTS, JsonExtensions.getJsonStringFromICacheRecordList(cacheRecords));
+            final String jsonString = JsonExtensions.getJsonStringFromICacheRecordList(cacheRecords);
+            try {
+                byte[] bytes = GzipUtil.compressString(jsonString);
+                Logger.info(TAG, "Get accounts, raw payload size :"
+                        + jsonString.getBytes().length + " compressed size " + bytes.length
+                );
+                resultBundle.putByteArray(BROKER_ACCOUNTS_COMPRESSED, bytes);
```

Comment:
How would this work with older MSAL? Do we need a version check to determine if we should compress?

Reply:
Yup, thanks for pointing out. I was planning to  address issue tomm  using our hello handshake, didn’t get time to look into. For now I got the test build to Teams  with these changes to test if it fixes the issue with new MSAL and new broker

Reply:
implemented here : https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/891/commits/6b19efd1418ec12e5e9adad0d81ee712f016700f